src/librustc/middle/free_region.rs

```diff
@@ -64,7 +64,7 @@ impl FreeRegionMap {
                     // No region bounds here
                 }
                 ty::Predicate::RegionOutlives(ty::Binder(ty::OutlivesPredicate(r_a, r_b))) => {
-                    match (r_a, r_b) {
+                    match (*r_a, *r_b) {
```

Comment:
Fair enough, I wanted to minimize the diff but I can go the other route too now that I know it works.